scenario_player/runner.py

```diff
@@ -153,279 +256,260 @@ def local_seed(self) -> str:
-        # Start root task
-        root_task_greenlet = gevent.spawn(self.root_task)
-        root_task_greenlet.name = "root_task"
-        greenlets.add(root_task_greenlet)
-        try:
-            gevent.joinall(greenlets, raise_error=True)
-        except BaseException as exc:
-            log.exception("Greenlet Error", exception=exc)
-            if not root_task_greenlet.dead:
-                # Make sure we kill the tasks if a node dies
-                root_task_greenlet.kill()
-            raise
+        return ether_transfer_transactions
 
-    def run_scenario(self):
-        mint_gas = GAS_LIMIT_FOR_TOKEN_CONTRACT_CALL * 2
+    def setup_mint_user_deposit_tokens_for_distribution(
+        self, node_addresses
+    ) -> Set[TransactionHash]:
+        """ Ensures the scenario player account has enough tokens and allowance
+        to fund the Raiden nodes.
+        """
+        orchestration_user_deposit_token_transactions: Set[TransactionHash] = set()
 
-        fund_tx, node_starter, node_addresses, node_count = self._initialize_nodes()
+        settings = self.definition.settings
+        udc_settings = settings.services.udc
+        balance_per_node = settings.services.udc.token.balance_per_node
 
-        node_monitor = self.node_controller.start_node_monitor()
-        greenlets = {node_monitor}
+        node_count = len(node_addresses)
+        required_allowance = balance_per_node * node_count
 
-        log.info("Node monitor started")
+        should_deposit_ud_token = udc_settings.token.deposit
+        if not udc_settings.enable or not should_deposit_ud_token:
+            return orchestration_user_deposit_token_transactions
 
-        setup = gevent.spawn(
-            self._setup, mint_gas, node_count, node_addresses, fund_tx, node_starter, greenlets
+        user_deposit_proxy, token_proxy = get_udc_and_corresponding_token(
+            udc_settings=udc_settings, settings=settings, proxy_manager=self.proxy_manager
         )
-        greenlets.add(setup)
 
-        gevent.joinall(greenlets, raise_error=True, count=1)
+        allowance_tx = userdeposit_maybe_increase_allowance(
+            token_proxy, user_deposit_proxy, required_allowance, maximum_allowance=UINT256_MAX
+        )
+        if allowance_tx:
+            orchestration_user_deposit_token_transactions.add(allowance_tx)
 
-    def _initialize_scenario_token(
-        self,
-        node_addresses: Set[ChecksumAddress],
-        udc_ctr: Optional[Contract],
-        should_deposit_ud_token: bool,
-        gas_limit: int,
+        mint_tx = token_maybe_mint(
+            token_proxy,
+            to_checksum_address(self.client.address),
+            minimum_balance=required_allowance,
+            maximum_balance=UINT256_MAX,
+        )
+        if mint_tx:
+            orchestration_user_deposit_token_transactions.add(mint_tx)
+
+        return orchestration_user_deposit_token_transactions
+
+    def setup_raiden_token_balances(
```

Comment:
This used to be in the middle of [`_initialize_nodes`](https://github.com/raiden-network/scenario-player/blob/f208ed6cb4f713c1efc921b10458365bcf918b5d/scenario_player/runner.py#L382-L405), now it is on its own method which allows for concurrent execution, and a clearer interface. (The rest of the code was managing nodes which has been grouped together).